packages/design-system/src/components/NativeDialog/NativeDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useRef, useEffect, useLayoutEffect } from 'react';
+import dialogPolyfill from 'dialog-polyfill';
+
+interface NativeDialogProps {
+  children: React.ReactNode;
+  /**
+   * Function called to close dialog.
+   */
+  exit: any;
+  /**
+   * Determines which dialog method is called.
+   * If true, `showModal()` is called, if false,
+   * `show()` is called.
+   */
+  showModal?: boolean;
+  /*
+   * @hide-prop Spreading props
+   */
+  [x: string]: any;
+}
+
+const NativeDialog = ({ children, exit, showModal, ...dialogProps }: NativeDialogProps) => {
+  const dialogRef = useRef(null);
+
+  // useLayoutEffect(() => {
+  //   if ((window as any).HTMLDialogElement === undefined) {
+  //     dialogPolyfill.registerDialog(dialogRef.current);
+  //   }
+  // })
+
+  useEffect(() => {
+    const dialogNode = dialogRef.current;
+    showModal ? dialogNode.showModal() : dialogNode.show();
+
+    return () => {
+      dialogNode.close();
+    };
+  }, [showModal]);
+
+  useEffect(() => {
+    const dialogNode = dialogRef.current;
+    const handleCancel = (event) => {
+      event.preventDefault();
+      exit();
+    };
+    dialogNode.addEventListener('cancel', handleCancel);
+    return () => {
+      dialogNode.removeEventListener('cancel', handleCancel);
```

Comment:
Chrome offers an `onCancel` event for dialog that's not fully supported. This fills in gaps for that for other browsers.